rotkehlchen/premium/premium.py

```diff
@@ -207,6 +212,78 @@ def set_credentials(self, credentials: PremiumCredentials) -> None:
             self.reset_credentials(old_credentials)
             raise PremiumAuthenticationError('rotki API key was rejected by server')
 
+    def get_remote_devices_information(self) -> dict:
+        """Get the list of devices for the current user"""
+        method = 'manage/premium/devices'
+        signature, data = self.sign(
+            method=method,
+            api_endpoint='webapi',
+        )
+        self.session.headers.update({'API-SIGN': base64.b64encode(signature.digest())})
+
+        try:
+            response = self.session.get(
+                f'{self.rotki_web}{method}',
+                data=data,
+                timeout=ROTKEHLCHEN_SERVER_TIMEOUT,
+            )
+        except requests.exceptions.RequestException as e:
+            msg = f'Could not connect to rotki server due to {e!s}'
+            log.error(msg)
+            raise RemoteError(msg) from e
+
+        data = _process_dict_response(response)
+        return data
+
+    def authenticate_device(self) -> None:
+        """
+        Check if the device is in the list of the devices and if it isn't add it when possible.
+        May raise:
+        - PremiumAuthenticationError: when the device can't be registered
+        """
+        device_data = self.get_remote_devices_information()
+        num_devices = len(device_data['devices'])
+        devices_limit = device_data['limit']
+        device_id = machineid.hashed_id(self.username)
+
+        device_found = False
+        for device in device_data['devices']:
+            if device['device_identifier'] != device_id:
+                continue
+            device_found = True
+            break
+
+        if not device_found:
+            if num_devices < devices_limit:
+                # try to sign up the device
+                self._register_new_device(device_id)
```

Comment:
```suggestion
                # try to register the device
                self._register_new_device(device_id)
```

Also question. How do you handle errors and show them to the user here? It's possible that the user uses a modified rotki trying to cheat, and the check above is skipped or the server logic has an error and then we just get the failure from the server

Reply:
This logic is used by `premium_create_and_verify` that already raises the possible errors `PremiumAuthenticationError` and `RemoteError`. For the frontend at the moment on error there is no custom message and the effect is a common error message saying that the premium couldn't be activated

Reply:
Still please take it out.